lib/ansible/module_utils/docker/swarm.py

```diff
@@ -166,6 +167,17 @@ def get_node_inspect(self, node_id=None, skip_missing=False):
 
         json_str = json.dumps(node_info, ensure_ascii=False)
         node_info = json.loads(json_str)
+
+        if 'ManagerStatus' in node_info:
+            if node_info['ManagerStatus'].get('Leader'):
+                # This is workaround of bug in Docker when in some cases the Leader IP is 0.0.0.0
+                # Check moby/moby#35437 for details
+                count_colons = node_info['ManagerStatus']['Addr'].count(":")
```

Comment:
A simple way to make it also work with IPv6 is the following:
```.py
addr = node_info['ManagerStatus']['Addr']
i = addr.rfind(':')
if i >= 0:
    addr = addr[:i]
if addr.startswith('[') and addr.endswith(']'):
    addr = addr[1:-1]
if addr:
    node_info['Status']['Addr'] = addr
```

Reply:
But there is no information the bug even exists when IPv6 is in use.